src/machine/machine_rp2040_uart.go

```diff
@@ -0,0 +1,132 @@
+// +build rp2040
+
+package machine
+
+import (
+	"device/rp"
+	"runtime/interrupt"
+)
+
+// UART on the RP2040.
+type UART struct {
+	Buffer    *RingBuffer
+	Bus       *rp.UART0_Type
+	Interrupt interrupt.Interrupt
+}
+
+func (uart *UART) Configure(config UARTConfig) error {
+	initUART(uart)
+
+	// Default baud rate to 115200.
+	if config.BaudRate == 0 {
+		config.BaudRate = 115200
+	}
+
+	// Use default pins if pins are not set.
+	if config.TX == 0 && config.RX == 0 {
+		// use default pins
+		config.TX = UART_TX_PIN
+		config.RX = UART_RX_PIN
+	}
+
+	uart.SetBaudRate(config.BaudRate)
+
+	// default to 8-1-N
+	uart.SetFormat(8, 1, ParityNone)
+
+	// Enable the UART, both TX and RX
+	uart.Bus.UARTCR.SetBits(rp.UART0_UARTCR_UARTEN |
+		rp.UART0_UARTCR_RXE |
+		rp.UART0_UARTCR_TXE)
+
+	// set GPIO mux to UART for the pins
+	config.TX.Configure(PinConfig{Mode: PinUART})
+	config.RX.Configure(PinConfig{Mode: PinUART})
+
+	// Enable RX IRQ.
+	uart.Interrupt.SetPriority(0x80)
+	uart.Interrupt.Enable()
+
+	// setup interrupt on receive
+	uart.Bus.UARTIMSC.Set(rp.UART0_UARTIMSC_RXIM)
```

Comment:
Looks like you're only supporting interrupts on UART0?

Reply:
The `UART0_UARTIMSC_RXIM` bitfield is generated and applies to either UART. 

`Bus` in this case being either `UART0` or `UART1`, and each UARTs has its own `UARTIMSC` register.